src/Hazelcast.Net/Serialization/Compact/Schema.cs

```diff
@@ -131,10 +127,16 @@ private void Initialize(string typeName, IEnumerable<SchemaField> typeFields)
             {
                 if (!fieldNames.Add(field.FieldName))
                     throw new ArgumentException($"Fields contain duplicate field name {field.FieldName}.", nameof(typeFields));
-                fieldsMap[field.FieldName] = field;
+                _fieldsMap[field.FieldName] = field;
             }
 
-            foreach (var field in fieldsMap.Values)
+            // the sorted set of fields, which will be used for fingerprinting - needs to be ordered
+            // exactly in the same way as Java, which uses Comparator.naturalOrder() i.e. "natural
+            // order", and good luck finding a definition for this, apart from it being case-sensitive,
+            // so we're going with whatever seems best in C# and hope it works.
+            Fields = _fieldsMap.Values.OrderBy(x => x.FieldName, StringComparer.InvariantCulture).ToArray();
```

Comment:
Can we reuse the typeFields variable here not to create another variable? 🤔 java stores whatever is passed to its constructor and then sorts it instead of creating a new.

Reply:
Problem is that `typeFields` is an un-ordered enumerable of fields, whereas `Fields` has to be ordered. In Java `fields` is a list of fields that we sort and capture, here `typeFields` is an `IEnumerable` which we cannot capture. Don't think it makes a big difference.